### PR TITLE
Show red budge on the more button till More screen is visible at least once

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #7019

### Description

The PR makes the red badge invisible after a user see the "more" screen at least once

### Testing instructions

- Open the app and notice that there is a red badge on the "more" button
- Click on the "more" button and notice that the red badge is gone

### Images/gif

https://user-images.githubusercontent.com/4923871/180968897-54169497-103a-4320-9c6d-cf2a24701d69.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt [hunk 1]
```diff
@@ -693,6 +695,13 @@ object AppPrefs {
     fun isPreLoginNotificationBeenDisplayed(): Boolean =
         getBoolean(DeletablePrefKey.PRE_LOGIN_NOTIFICATION_DISPLAYED, false)
 
+    fun isUserSeenNewFeatureOnMoreScreen(): Boolean =
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt:698]: ❓ What do you think about writing a test for this?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuNewFeatureAvailableProvider.kt [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
+import kotlinx.coroutines.flow.map
 import javax.inject.Inject
 
 @Reusable
-class MoreMenuNewFeatureProvider @Inject constructor() {
-    fun provideNewFeatures() = listOf(Payments)
+class MoreMenuNewFeatureHandler @Inject constructor(
+    private val appPrefsWrapper: AppPrefsWrapper,
+) {
+    val moreMenuNewFeaturesAvailable = appPrefsWrapper.observePrefs()
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuNewFeatureAvailableProvider.kt]: Nice 👏

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuNewFeatureAvailableProvider.kt [hunk 3]
```diff
@@ -1,12 +1,27 @@
...
+import com.woocommerce.android.AppPrefsWrapper
 import com.woocommerce.android.ui.moremenu.MoreMenuNewFeature.Payments
 import dagger.Reusable
+import kotlinx.coroutines.flow.map
 import javax.inject.Inject
 
 @Reusable
-class MoreMenuNewFeatureProvider @Inject constructor() {
-    fun provideNewFeatures() = listOf(Payments)
+class MoreMenuNewFeatureHandler @Inject constructor(
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuNewFeatureAvailableProvider.kt]: ❓ File name and the class name are different. Just wanted to check if this is intentional.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuViewModelTests.kt [hunk 4]
```diff
@@ -67,4 +70,14 @@ class MoreMenuViewModelTests : BaseUnitTest() {
 
         assertThat(states.size).isEqualTo(2)
     }
+
+    @Test
+    fun `when on view resumed, then new feature handler marks new feature as seen`() = testBlocking {
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuViewModelTests.kt:75]: 🔍 nitpick: I think the given, and when are a little mixed up here. Should it be like this?
```
//Given
setup { }

//When
viewModel.onViewResumed()

//Then
verify(moreMenuNewFeatureHandler).markNewFeatureAsSeen()
```

**Review by AnirudhBhat — APPROVED** [review]
Thank you for working on this @kidinov I've tested the changes and it works as expected. I've left a few very minor comments and none of them are blocking.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt:698]: Done e37ef73e5227c3dcbeecb5cdda46787b6bcf2127

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuNewFeatureAvailableProvider.kt (hunk 3 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuNewFeatureAvailableProvider.kt]: Nope, it's a mistake. Fixed here - dc5157f8865a15105ec9c059445d2b7ff98ef783

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuViewModelTests.kt (hunk 4 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuViewModelTests.kt:75]: yep, fixed here - 81c21e3fd66c1d8c392885f6de9f38e6b9769f41

**Pull request merged**